sys/net/application_layer/dhcpv6/client.c

```diff
@@ -247,6 +247,17 @@ static inline size_t _compose_elapsed_time_opt(dhcpv6_opt_elapsed_time_t *time)
     return len + sizeof(dhcpv6_opt_t);
 }
 
+static inline size_t _compose_mud_url_opt(dhcpv6_opt_mud_url_t *mud_url_opt,
+                                          const char *mud_url)
+{
+    uint16_t len = strlen(mud_url);
+
```

Comment:
```suggestion
    if (len > len_max) {
        assert(0);
        return 0;
    }
```

This way if the configuration is wrong, it will crash immediately - I'd say that's preferable to `DEBUG()` output that might be disabled and `printf()` warnings that take up space in the final binary.

Reply:
Thank you! That solves the problem really well. I also think that crashing is the best option here as one goal of MUD is to improve the security within a network and sending a corrupted URL would clearly undermine this intention. 

I added your suggestions in commit 70391d8 .